Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs

```diff
@@ -353,20 +353,27 @@ protected void SetupEquipmentInfo()
                 // We do not have item properties and have growing stats! So don't display a finished stat but a range instead.
                 else
                 {
-                    _ = mItem.TryGetRangeFor((Stat)i, out var range);
-                    var statLow = mItem.StatsGiven[i] + range.LowRange;
-                    var statHigh = mItem.StatsGiven[i] + range.HighRange;
-
-                    if (mItem.PercentageStatsGiven[i] != 0)
-                    {
-                        rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], Strings.ItemDescription.RegularAndPercentage.ToString(Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh), mItem.PercentageStatsGiven[i]));
-                    }
-                    else
+                    if (mItem.TryGetRangeFor((Stat)i, out var range))
                     {
-                        rows.AddKeyValueRow(Strings.ItemDescription.StatCounts[i], Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh));
+                        var statGiven = mItem.StatsGiven[i];
+                        var percentageStatGiven = mItem.PercentageStatsGiven[i];
+                        var statLow = statGiven + range.LowRange;
+                        var statHigh = statGiven + range.HighRange;
+
+                        string statMessage;
+
+                        if (percentageStatGiven != 0)
+                        {
+                            statMessage = Strings.ItemDescription.RegularAndPercentage.ToString(Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh), percentageStatGiven);
+                        }
+                        else
+                        {
+                            statMessage = Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh);
```

Comment:
Correct me if I'm wrong, but I think I'm seeing `Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh)` computed in both arms of the if statement?

So you can instead do `var statMessage = Strings.ItemDescription.StatGrowthRange.ToString(statLow, statHigh);` before the if statement, and do `statMessage = ....ToString(statMessage, percentageStatGiven)` in the `if` arm (delete the `else` arm).

Reply:
You're right, thanks